server/raft_transport.go

```diff
@@ -116,28 +117,42 @@ func (t *rpcTransport) Send(req *multiraft.RaftMessageRequest) error {
 		return err
 	}
 
-	client := rpc.NewClient(addr, nil, t.rpcContext)
-	select {
-	case <-client.Ready:
-	case <-client.Closed:
-		return util.Errorf("raft client failed to connect")
-	}
-
-	call := client.Go(raftMessageName, protoReq, &proto.RaftMessageResponse{}, nil)
-	select {
-	case <-call.Done:
-		// If the call failed synchronously, report an error.
-		return call.Error
-	default:
-		// Otherwise, fire-and-forget.
-		go func() {
-			<-call.Done
-			if call.Error != nil {
-				log.Errorf("raft message failed: %s", call.Error)
-			}
-		}()
-		return nil
-	}
+	// NewClient can block, so fire-and-forget from here on.
+	// TODO(tschottdorf): An unbounded number of goroutines may be launched
+	// here. We should put a closer for each goroutine in a list and limit
+	// the number of concurrent requests so that the oldest request gets
+	// killed when a threshold is reached.
+	go func() {
+		deadline := time.After(raftMessageTimeout)
+		client := rpc.NewClient(addr, nil, t.rpcContext)
+		select {
+		case <-client.Closed:
+			log.Errorf("raft client failed to connect")
```

Comment:
Add a TODO to introduce some sort of error channel so we can pass the error back to multiraft so it can call ReportUnreachable.